### PR TITLE
New package: fritzing

### PR DESCRIPTION
**Pull request by jeremyd2019**
My first attempt at packaging something.

> The Fritzing application is an Electronic Design Automation software with a low entry barrier, suited for the needs of makers and hobbyists. It offers a unique real-life "breadboard" view, and a parts library with many commonly used high-level components. Fritzing makes it very easy to communicate about circuits, as well as to turn them into PCB layouts ready for production. It is particularly popular among Arduino and Raspberry Pi users, and is widely used in education and creative tinkering.

https://fritzing.org
https://github.com/fritzing/fritzing-app

Major ~~outstanding~~[solved](https://github.com/msys2/MINGW-packages/pull/7376#issuecomment-738217175) issue is that Fritzing cannot find its files out of the box.  This can be worked around by launching `Fritzing -f /mingw{32,64}/share/fritzing`.  It passes path defines to the compiler, but I found that in order for `qmake` to allow `make install` to work properly I had to use `MSYS2_ARG_CONV_EXCL="PREFIX="`.  That results in that define using a path starting with /mingw64.  Not that the alternative would have been better, since that was an absolute path on the GHA builder...  Also, the code that uses that is ifdef'd out on `QT_OS_WIN`.  Is there some common solution for mingw programs to know their prefix, so they can find their installed files?

A couple of points of oddness about this package:
* Fritzing is no longer officially supported on 32-bit.  However, it does still build and seems to run fine.
* Fritzing keeps its parts database in a git repository, and uses libgit2 to keep it up to date.  This resulted in a bit of a song-and-dance with `git` to turn the `fritzing-parts` checkout from makepkg into a self-contained git repo with the proper origin url and branch name.

### CODE REVIEW

**Comment by Biswa96** [comment]
> Is there some common solution for mingw programs to know their prefix

I am not sure about your desired solution. Extrapolating from the `/mingw{32,64}/share/fritzing` path, I think mingw-w64-mdloader\01-get-applet-path.patch file may help here.

**Comment by Biswa96** [comment]
> Is there some common solution for mingw programs to know their prefix

I am not sure about your desired solution. Extrapolating from the `/mingw{32,64}/share/fritzing` path, I think `mingw-w64-mdloader/01-get-applet-path.patch` file may help here. BTW, if the patches are in upstream repository a little comment with the commit hash will be useful.

**Comment by jeremyd2019** [comment]
> I am not sure about your desired solution. Extrapolating from the `/mingw{32,64}/share/fritzing` path, I think `mingw-w64-mdloader/01-get-applet-path.patch` file may help here.

Yeah, relative to the executable was kind of what I was thinking, too...

> BTW, if the patches are in upstream repository a little comment with the commit hash will be useful.

One was cherry-picked.  Comment added.  I'm planning to submit the other numbered patches as a pull request upstream, for mingw support.

**Comment by jeremyd2019** [comment]
The patch switching to the unix behavior for looking for files on mingw, and adding `../share/fritzing` relative to to the executable's location as a candidate, seems to work well.  Also, I noticed that calling `uname -m` from qmake to determine whether it was a 32 or 64-bit build did not work correctly, despite my recollection that uname output was changed based on MSYSTEM.  I switched it to use the $MSYSTEM_CARCH environment variable instead.

File: mingw-w64-fritzing/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+arch=('any')
+url="https://github.com/fritzing/fritzing-app"
+license=('GPL' 'CCPL:cc-by-sa-3.0')
+makedepends=("${MINGW_PACKAGE_PREFIX}-boost"
+             'git')
+depends=("${MINGW_PACKAGE_PREFIX}-qt5"
+         "${MINGW_PACKAGE_PREFIX}-libgit2"
+         "${MINGW_PACKAGE_PREFIX}-zlib")
+source=("${_realname}"::"git+https://github.com/fritzing/fritzing-app.git#tag=CD-498"
+        "${_realname}-parts"::"git+https://github.com/fritzing/fritzing-parts.git#branch=master"
```
Comment (lazka) [inline mingw-w64-fritzing/PKGBUILD:18]: needs a fixed commit

**Pull request merged**